pkg/module/module_test.go

```diff
@@ -0,0 +1,30 @@
+package module
```

Comment:
I do not want pkg/module to be a part of the public API. I would prefer it if the Flow controller hid the implementation of the interfaces.

Reply:
Moved this into flow, which required an abstraction for flow api, which was something long overdue since it kept tripping things up. Only module interfaces are exposed publicly now